src/internal_events/splunk_hec.rs

```diff
@@ -35,6 +36,27 @@ impl InternalEvent for SplunkEventEncodeError {
     }
 }
 
+#[derive(Debug)]
+pub(crate) struct SplunkInvalidMetricReceived<'a> {
+    pub value: &'a MetricValue,
+    pub kind: &'a MetricKind,
+}
+
+impl<'a> InternalEvent for SplunkInvalidMetricReceived<'a> {
+    fn emit_logs(&self) {
+        warn!(
+            message = "Invalid metric received; dropping event.",
+            value = ?self.value,
+            kind = ?self.kind,
+            internal_log_rate_secs = 30,
+        )
+    }
+
+    fn emit_metrics(&self) {
+        counter!("processing_errors_total", 1, "error_type" => "invalid_metric");
```

Comment:
```suggestion
        counter!("processing_errors_total", 1, "error_type" => "invalid_metric_kind");
```

Could make it a bit more clear.